### PR TITLE
app.py supports restful-api

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

File: docs/zh_cn/restful_api.md [hunk 1]
```diff
@@ -107,6 +107,14 @@ curl http://{server_name}:{server_port}/v1/embeddings \
   }'
 ```
 
+### webui
+
+也可以直接用 webui 测试使用 restful-api。
+
+```shell
+python -m lmdeploy.serve.gradio.app restful_api_url server_name --restful_api True
```
Comment (lvhan028) [inline docs/zh_cn/restful_api.md:115]: Can `server_name` be put to `restful_api_url`? May provide an example above this command as a comment

**Review by tpoisonooo — COMMENTED** [review]
also update README.md

File: lmdeploy/serve/openai/api_client.py [hunk 2]
```diff
@@ -6,14 +6,24 @@
 import requests
```
Comment (tpoisonooo) [inline lmdeploy/serve/openai/api_client.py:6]: update requirements.txt   `shortuuid`

File: docs/en/restful_api.md [hunk 3]
```diff
@@ -6,7 +6,7 @@
 python3 -m lmdeploy.serve.openai.api_server ./workspace server_name server_port --instance_num 32 --tp 1
```
Comment (tpoisonooo) [inline docs/en/restful_api.md:6]: update server_name to 0.0.0.0, I'm afraid that some guy would give 127.0.0.1

File: docs/en/restful_api.md [hunk 4]
```diff
@@ -6,7 +6,7 @@
...
 ```
 
-Then, the user can open the swagger UI: http://{server_name}:{server_port}/docs for the detailed api usage.
```
Comment (tpoisonooo) [inline docs/en/restful_api.md:9]: `server_name`  to `server_ip`

**Comment by tpoisonooo** [comment]
About  `Rust/Golang`,   give this link https://github.com/swagger-api/swagger-codegen  would be enough.

convert `http://10.140.24.141:19999/openapi.json` to  java/rust/gloang client, not  http client usage.

**Comment by lvhan028** [comment]
No blanks in the generated text. 
It's 100% reproducible using llama-2-7b-chat model

```
double enter to end input >>> hi

Hello!It'snicetomeetyou.I'mheretohelpyouwithanyquestionsorconcernsyoumayhave.Pleasefeelfreetoaskmeanything,andIwilldomybesttoprovidehelpfulandaccurateinformation.I'mprogrammedtoberespectful,honest,andsociallyunbiased,soyoucantrustthatmyresponseswillbefreefromharmfuloroffensivecontent.Ifaquestiondoesn'tmakesenseorisnotfactuallycoherent,Iwillexplainwhyinsteadofprovidinganincorrectanswer.AndifIdon'tknowtheanswertoaquestion,Iwillletyouknowratherthanprovidingfalseinformation.Pleasegoaheadandaskmeanything!

```

**Review by lvhan028 — APPROVED** [review]

**Review by tpoisonooo — APPROVED** [review]

**Comment by tpoisonooo** [comment]
typo 直接给你改了，git pull 一下。

**Pull request merged**